libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -389,8 +390,15 @@ int32_t AP_Mission::get_next_ground_course_cd(int32_t default_angle)
 }
 
 // set_current_cmd - jumps to command specified by index
-bool AP_Mission::set_current_cmd(uint16_t index)
+bool AP_Mission::set_current_cmd(uint16_t index, bool set_jump_to_land)
```

Comment:
Why wouldn't we always want to reset it in this case? It's a bit easier to track, and protects against the AFS cases for example which are currently broken.

Reply:
Agreed.  I have removed the set_jump_to_land bool.  Tested in SITL and the behaviour remains unchanged.